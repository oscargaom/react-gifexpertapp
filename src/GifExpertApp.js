import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import PropTypes from 'prop-types';

export const GifExpertApp = ({defaultCategories = []}) => {

    //const categorias = ['One Punch'];
    const [categories, setCategories] = useState(defaultCategories)
    // const [categories2, setCategories2] = useState([1])
    // const [categories3, setCategories3] = useState([2])

    // const handlerAdd = () => {
    //     console.log(categories);

    //     // setCategories( cat => [...cat, 'Nuevo']);
    // };

    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>
            {/* <button onClick={ handlerAdd }>Agregar</button> */}
            <ol>
                {
                    categories.map( cat => (
                        //  <li key={cat}>{cat}</li>
                        <GifGrid key={cat} category={cat}/>
                    ))
                }
            </ol>

        </div>
    )
}

GifExpertApp.propTypes = {
    defaultCategories: PropTypes.array
}