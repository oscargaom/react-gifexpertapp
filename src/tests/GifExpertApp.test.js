import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas unitarias del componente <GifExpertApp />', () => {

    const categories = ['Goku', 'Homero'];

    test('Validar que el componete exista ', () => {
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Validar que el número de categorías corresponde al número de componentes', () => {
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
    })
    
})
