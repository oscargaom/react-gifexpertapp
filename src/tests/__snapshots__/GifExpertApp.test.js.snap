// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias del componente <GifExpertApp /> Validar que el componete exista  1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="Homero"
      key="Homero"
    />
  </ol>
</div>
`;

exports[`Pruebas unitarias del componente <GifExpertApp /> Validar que el número de categorías corresponde al número de componentes 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="Homero"
      key="Homero"
    />
  </ol>
</div>
`;
